client/src/main/proto/primitive.proto

```diff
@@ -72,34 +74,73 @@ enum RolePermission {
   can_get_all_txs = 36;
   can_get_blocks = 42;
   can_get_peers = 45;
+  can_get_my_engine_receipts = 50;
+  can_get_domain_engine_receipts = 51;
+  can_get_all_engine_receipts = 52;
 
   // Grant permissions
   can_grant_can_set_my_quorum = 37;
   can_grant_can_add_my_signatory = 38;
   can_grant_can_remove_my_signatory = 39;
   can_grant_can_transfer_my_assets = 40;
   can_grant_can_set_my_account_detail = 41;
+  can_grant_can_call_engine_on_my_behalf = 49;
+
+  // Root Permission
+  root = 47;
 }
 
 enum GrantablePermission {
   can_add_my_signatory = 0;
   can_remove_my_signatory = 1;
   can_set_my_quorum = 2;
   can_set_my_account_detail = 3;
-  can_transfer_my_assets = 4; // not implemented now
+  can_transfer_my_assets = 4;  // not implemented now
```

Comment:
I think it is implemented 🤔

Reply:
It is copy of Iroha file, reported to Iroha.